### PR TITLE
Update component to use provision-react-compoment

### PR DESCRIPTION
**Pull request by danielmoll**
Get it working with `provision-react-component` and start to prepare the gallery to be more modular for use with Elections Mythbuster, for example.

This is a BREAKING CHANGE.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  handleSlideChange(...args) {
+    const first = 0;
+    const index = args[first];
+    this.setState({ activeSlide: index });
+  }
+
+  renderGalleryImage(image, alt, index) {
+    const { activeSlide } = this.state;
+    let klass = 'gallery__slide gallery__slide--hidden';
```
Comment (gcedo) [inline src/index.js]: ternary op?

Comment (keithamus) [inline src/index.js]: `klass` is probably not the best name for this - it would be better to be `className`.

File: src/example.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import Gallery from './';
+import React from 'react';
+import images from '../data.json';
+
+export default (
+  <Gallery images={images} />
+);
```
Comment (keithamus) [inline src/example.js]: I know this PR didn't instigate this - but it doesn't really make for the best example, deferring the core object structure to some other file

**Comment by danielmoll** [comment]
@keithamus @gcedo updated :)

**Comment by keithamus** [comment]
LGTM; I'll leave @gcedo to merge

**Comment by danielmoll** [comment]
@gcedo would you mind merging if all ok? :)

**Pull request merged**